src/model/reference.ml

```diff
@@ -378,13 +470,21 @@ let parse whole_reference_location s :
               location
             |> Error.raise_exception
         | `TPage ->
-            let suggestion =
-              Printf.sprintf "'page-%s' should be first." identifier
+            let () =
+              match next_token.kind with
+              | `End_in_slash -> ()
+              | `None | `Prefixed _ ->
+                  let suggestion =
+                    Printf.sprintf "Reference pages as '<parent_path>/%s'."
+                      identifier
+                  in
+                  not_allowed ~what:"Page label"
+                    ~in_what:"on the right side of a dot" ~suggestion location
+                  |> Error.raise_exception
             in
-            not_allowed ~what:"Page label"
-              ~in_what:"the last component of a reference path" ~suggestion
-              location
-            |> Error.raise_exception)
+            (* Prefixed pages are not differentiated. *)
+            `Page_path (path [ identifier ] next_token tokens)
+        | `TPathComponent -> `Page_path (path [ identifier ] next_token tokens))
```

Comment:
Same:
> Do I understand correctly that this can only happen in case of `/` ending references, such as `{!/a/b/}`? If so, I think that those references are currently not allowed.
>
> If we allow them, shouldn't it be `` `Any_path ``? :)

Reply:
Same fix.

Reply:
Hmm it seems the `assert false` in this case was forgotten!

Reply:
That's right! In fact none of the `` `TPathComponent `` cases are reachable, path components can only be parsed by the `path` function.